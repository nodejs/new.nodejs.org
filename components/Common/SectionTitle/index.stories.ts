import SectionTitle from './index';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type Story = StoryObj<typeof SectionTitle>;
type Meta = MetaObj<typeof SectionTitle>;

export const Default: Story = {
  args: {
    path: ['home', 'previous', 'current'],
  },
};

export default { component: SectionTitle } as Meta;
