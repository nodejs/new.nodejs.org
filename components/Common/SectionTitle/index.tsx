import styles from './index.module.scss';
import type { FC } from 'react';

type SectionTitleProps = { path: string[] };

const SectionTitle: FC<SectionTitleProps> = ({ path }) => (
  <div className={styles.sectionTitle}>
    {path.map((item, index) => {
      const isLast = index === path.length - 1;

      if (isLast) {
        return (
          <span className={styles.active} key={item}>
            {item}
          </span>
        );
      }

      return `${item} / `;
    })}
  </div>
);

export default SectionTitle;
