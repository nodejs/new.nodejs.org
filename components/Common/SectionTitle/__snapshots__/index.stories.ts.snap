// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common/SectionTitle Default smoke-test 1`] = `
<div class="SectionTitle_sectionTitle__mNok7">
  home / previous /
  <span class="SectionTitle_active__XRLOn">
    current
  </span>
</div>
`;
